puma/matshow.py

```diff
@@ -0,0 +1,198 @@
+        colormap: plt.cm = plt.cm.Oranges,
+        cbar_label: str | None = None,
+        atlas_offset: float = 1,
+        **kwargs,
+    ) -> None:
+        """Plot a matrix with matplotlib matshow.
+
+        Parameters
+        ----------
+        matrix : np.ndarray
+            The matrix to be plotted.
+        x_ticklabels : list | None, optional
+            Names of the matrix's columns; if None, indices are shown. by default None
+        x_ticks_rotation : int, optional
+            Rotation of the columns' names, by default 90
+        y_ticklabels : list | None, optional
+            Names of the matrix's rows; if None, indices are shown. by default None
+        show_entries : bool, optional
+            If True, show matrix entries as numbers in the matrix pixels. by default True
+        show_percentage : bool, optional
+            If True, if matrix entries are percentages (i.e. numbers in [0,1]), format them as
+            percentages. by default False
+        text_color_threshold : float, optional
+            threshold on the relative luminance of the colormap bkg color after which the text color
+            switches to black, to allow better readability on lighter cmap bkg colors.
+            If 1, all text is white; if 0, all text is black. by default 0.408
+        colormap : plt.cm, optional
+            Colormap for the plot, by default `plt.cm.Oranges`
+        cbar_label : str | None, optional
+            Label of the colorbar, by default None
+        atlas_offset : float, optional
+            Space at the top of the plot reserved to the Atlasify text. by default 1
+        **kwargs : kwargs
+            Keyword arguments for `puma.PlotObject`
+
+        Example
+        -------
+        >>> mat = np.random.rand(4, 3)
+        >>> plot_mat = MatshowPlot(mat)
+        """
+        super().__init__(**kwargs)
+
+        # Checking size consistency of ticklabels
+        if x_ticklabels is not None:
+            assert (
+                len(x_ticklabels) == matrix.shape[1]
+            ), "MatshowPlot: mismatch between x_tickslabels and number of columns in the matrix."
+
+        if y_ticklabels is not None:
+            assert (
+                len(y_ticklabels) == matrix.shape[0]
+            ), "MatshowPlot: mismatch between y_tickslabels and number of rows in the matrix."
+
+        self.mat = matrix
+        self.x_ticklabels = x_ticklabels
+        self.x_ticks_rotation = x_ticks_rotation
+        self.y_ticklabels = y_ticklabels
+        self.show_entries = show_entries
+        self.show_percentage = show_percentage
+        self.text_color_threshold = text_color_threshold
+        self.colormap = colormap
+        self.cbar_label = cbar_label
+        self.atlas_offset = atlas_offset
+
+        # Specifying figsize if not specified by user
+        if self.figsize is None:
+            self.figsize = (10, 10.5)
+        self.initialise_figure()
+        self.__plot()
```

Comment:
Just to be sure: are you suggesting to make the `plot` method public so that the user can call it after initializing the `MatshowPlot` class? I based this code on the `PiePlot` class in `pie.py`, where the (public) method `plot` is automatically called by the class constructor.
But, if you prefer the first approach, I'm happy to modify the code (also, I think is a bit cleaner that way, since it allows to declare one `MatshowPlot` class with a defined appearance and use it to plot different matrixes by calling `plot` multiple times).

Reply:
My suggestion was to add a public `draw` method, as we have [here](https://github.com/umami-hep/puma/blob/main/puma/var_vs_var.py#L385), for example. And when we want to draw, we call it.

However, I am unsure about such an implementation.

Reply:
Thanks, I see what you mean :) I'll wait for Sam's opinion and then I'll modify the code in whichever way you think works best!

Reply:
Sounds like a good idea to keep the approach consistent with other plots :) thanks

Reply:
@samvanstroud @dkobylianskii I modified the code accordingly, now I think we should be ready :)